### ISSUE TITLE
Multipass can't get instances' IPs on macOS Sequoia

### ISSUE THREAD

**Issue by naynayu** [canonical/multipass#3661]
**Describe the bug**
I have been trying to create instances for a few weeks but it always times out. I even uninstalled and reinstalled, same issue. I have attached logs, hopefully, someone is able to make sense of it. 

**To Reproduce**
How, and what happened?
1. `multipass launch`
2. Process starts and never completes

**Expected behavior**
What did you expect to happen?

I expect that if launch a VM it should start and launch a VM both on the terminal and open a shell on the GUI

**Logs**
Please provide logs from the daemon, see [accessing logs](https://multipass.run/docs/accessing-logs) on where to find them on your platform.

The logs attached show what happens when I try to spin up an instance on both CLI and GUI at the same time
[Multipass Logs 4_9.txt](https://github.com/user-attachments/files/16858668/Multipass.Logs.4_9.txt)

**Additional info**
 - OS: macOS Sequoia Beta 15.0 (24A5331b)
- `multipass version`
multipass   1.14.0+mac
multipassd  1.14.0+mac
- `multipass info`
Name:           adapted-bunny
State:          Unknown
Snapshots:      0
IPv4:           --
Release:        --
Image hash:     f4b47dd0a608 (Ubuntu 20.04 LTS)
CPU(s):         --
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --

Name:           sporting-salmon
State:          Unknown
Snapshots:      0
IPv4:           --
Release:        --
Image hash:     5ecac6447be6 (Ubuntu 24.04 LTS)
CPU(s):         --
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --

Name:           wired-vole
State:          Unknown
Snapshots:      0
IPv4:           --
Release:        --
Image hash:     5ecac6447be6 (Ubuntu 24.04 LTS)
CPU(s):         --
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --

- `multipass get local.driver`
qemu
**Additional context**
<img width="354" alt="image" src="https://github.com/user-attachments/assets/fb29970e-3d5a-4a99-9da3-7dd2c670bcb7">
<img width="443" alt="image" src="https://github.com/user-attachments/assets/d87f62be-4ac2-4b29-a8ae-0478b872c6b1">

**Comment by Andrey-DevOps** [comment]
The same issue

**Comment by ricab** [comment]
Hi @Naynayu, do you have your firewall turned on? Does the situation improve if you disable it? 

BTW, you appear to have attached GUI logs. We'd need the Multipass daemon's logs instead.

**Comment by ricab** [comment]
Hi @Naynayu, does `ls /Library/Logs/Multipass/multipassd.log` not find anything? But yes, Multipass logs from the console should be the same thing.

Be sure to have a look at our [troubleshooting doc](https://multipass.run/docs/troubleshoot-networking#troubleshoot-networking-on-macos). We are currently working to update it, but it has some important info as it is.

**Comment by Naynayu** [comment]
@ricab So ls /Library/Logs/Multipass/multipassd.log does not produce anything. But when I did `cat /Library/Logs/Multipass/multipassd.log,` it worked. Here are the logs:

[Multipassd.txt](https://github.com/user-attachments/files/16890521/Multipassd.txt)

**Comment by Naynayu** [comment]
Adding firewall outputs as well:

![image](https://github.com/user-attachments/assets/8104242b-4932-47a6-915d-a0972dcbe8a4)

**Comment by Naynayu** [comment]
@ricab Hi, I have tried all the solutions on the troubleshooting doc, no luck. 😭

**Comment by ricab** [comment]
Hi @nayneenarh, I just had a look at the logs you sent and I see two cases of instances starting up and having their network interfaces up, but Multipass not being able to SSH in. Most likely, the instance isn't getting an IP or Multipass can't find the right one. We are trying to update our troubleshooting but in the meantime please have a look at [this recipe](https://github.com/canonical/multipass/discussions/3660#discussioncomment-10548808). You should be in either case 2 or 3. Let us know how you fare with that.

**Comment by IronCore864** [comment]
It was working in previous versions but stopped working on macOS Sequoia Beta (and on the official release today, too). Probably something related to the new OS version.

**Comment by zamb0** [comment]
Hey @nayneenarh, same issue 🥲

**Comment by ricab** [comment]
Hi all, it looks like we might have a problem in Sequoia, which we are investigating. Sorry for your troubles.

**Comment by ricab** [comment]
FYI, we have found the issue. Now we need to find a solution.

**Comment by nayneenarh** [comment]
> FYI, we have found the issue. Now we need to find a solution.

Oh great! Looking forward to the solution. If it's something we can also help with, please feel free to let us know.

**Comment by vsarunas** [comment]
Now that the RC has been released; it appears that the VM is actually up; and IP address is assigned, just Multipass is not able to get that. For example shutdown causes logs like:
```
Could not broadcast shutdown message in VM: ssh connection failed: 'Timeout connecting to 192.168.67.6'
```

While new VMs starting have:
```
[trace] [immune-flathead] Could not determine IP address within 1000ms
```

The VM is actually up and running as shown by `arp` output; a little script to parse the IPs:
```
#!/bin/bash
multipass list | grep -e "Unknown" -e "Starting" | awk '{print $1}' | while read -r vm; do
    mac=$(ps -ef | grep "qemu-system" | grep "$vm" | grep -o 'mac=[0-9a-fA-F:]*' | awk -F= '{print $2}')
    [ -z "$mac" ] && mac="unknown"
    # convert :00: to :0: or :0a: to :a: several times
    for i in {1..6}; do
        mac=$(echo "$mac" | sed -E 's/:0([0-9a-fA-F]):/:\1:/g; s/:00:/:0:/g')
    done
    
    ip=$(arp -an | grep "$mac" | awk -F'[()]' '{print $2}')  
    printf "VM: %-25s  IP: %-15s MAC Address: %-17s\n" "$vm" "${ip:-unknown}" "$mac"
done
```

```
$ multipass list
Name                    State             IPv4             Image
accepting-planthopper   Unknown           --               Ubuntu 24.04 LTS
immune-flathead         Unknown           --               Ubuntu 24.04 LTS
linux-dev               Unknown           --               Ubuntu 22.04 LTS
```

```
$ bash multipass_ip.sh 
VM: accepting-planthopper      IP: 192.168.67.9    MAC Address: 52:54:0:1e:18:4e 
VM: immune-flathead            IP: 192.168.67.8    MAC Address: 52:54:0:d:4a:22  
VM: linux-dev                  IP: 192.168.67.7    MAC Address: 52:54:0:14:e0:76 
```

I'm able to create a new VM and login to it using root's key:
```
ssh -i /var/root/Library/Application\ Support/multipassd/ssh-keys/id_rsa ubuntu@192.168.67.9
```

**Comment by vsarunas** [comment]
It appears that the format of `/var/db/dhcpd_leases` has changed or is corrupted on new version; previously it had `1,mac_address`, now something is very long and doesn't respond to IP instance actually gets assigned:

```
more /var/db/dhcpd_leases
{
        name=accepting-planthopper
        ip_address=192.168.67.4
        hw_address=ff,f1:f5:dd:7f:0:2:0:0:ab:11:32:ca:56:11:8a:74:67:42
        identifier=ff,f1:f5:dd:7f:0:2:0:0:ab:11:32:ca:56:11:8a:74:67:42
        lease=0x66e8a656
}
{
        name=immune-flathead
        ip_address=192.168.67.3
        hw_address=ff,f1:f5:dd:7f:0:2:0:0:ab:11:b4:f8:94:34:d8:4b:79:9c
        identifier=ff,f1:f5:dd:7f:0:2:0:0:ab:11:b4:f8:94:34:d8:4b:79:9c
        lease=0x66e8a642
}
```

**Comment by georgeliao** [comment]
@vsarunas 
Thanks for sharing your findings. Meanwhile, we also found the exact same thingsand we are working on a more robust solution which can retrieve the ip address from the `/var/db/dhcpd_leases` file.

**Comment by asemarian** [comment]
I'm having the same problem today after upgrading to macOS Sequoia last night.

**Comment by geozeke** [comment]
I'll add, me too (upgraded to Sequoia yesterday), but this information may help: If I launch an instance using a bridged network (`--bridged`) it still hangs on "Starting..." and eventually times out. However it pops up, by name, on my network and gets a valid IP address according to my router portal.

**Comment by johalun** [comment]
> I'll add, me too (upgraded to Sequoia yesterday), but this information may help: If I launch an instance using a bridged network (`--bridged`) it still hangs on "Starting..." and eventually times out. However it pops up, by name, on my network and gets a valid IP address according to my router portal.

Confirmed this works for a new VM using the launch command, but how would do get this to work for an existing VM?

**Comment by geozeke** [comment]
Thanks. This is curious. Even though my bridged network attempt still results in the VM's IP popping up on my network, the VM itself never starts. Remains stuck on `Starting...`, then times out.

It looks like the team is zeroing in on fix. I'll wait for bit. 😉

**Comment by johalun** [comment]
@freef4ll Thanks, but the problem was never finding the IP. I did that using `nmap` to scan for open ssh ports on the VM subnet. The problem was that the disk  got corrupted by Multipass so the VM was stuck in initramfs, requiring manual `fsck`.

**Comment by kacperk** [comment]
It seems that multipass is missing system permissions to connect to local networks (I have a pop-up to allow permissions to connect to local network for many apps since upgrade to Sequoia, but not from multipass). I can connect to vm via ssh, I can ping it from terminal, but multipass itself can't access it.

**Comment by digows** [comment]
> It seems that multipass is missing system permissions to connect to local networks (I have a pop-up to allow permissions to connect to local network for many apps since upgrade to Sequoia, but not from multipass). I can connect to vm via ssh, I can ping it from terminal, but multipass itself can't access it.

Same page, same thoughts.

**Comment by ricab** [comment]
Hi everyone. We have a prototype of a fix that you can install from [this test package](https://multipass-ci.s3.amazonaws.com/pr661/multipass-1.15.0-dev.2929.pr661%2Bgc67ef6641.mac-Darwin.pkg). Let us know how it fares for you!

**Comment by shaunabanana** [comment]
Just tried the fix and it works! Multipass now asks for local network access permission when first starting an instance. Thank you!

> Hi everyone. We have a prototype of a fix that you can install from [this test package](https://multipass-ci.s3.amazonaws.com/pr661/multipass-1.15.0-dev.2929.pr661%2Bgc67ef6641.mac-Darwin.pkg). Let us know how it fares for you!

**Comment by asemarian** [comment]
Hey @ricab, I'm happy to report that it works! Gosh, it's been a rough week. Thank you guys for releasing a fix so quickly 🙏

**Comment by ricab** [comment]
Great, thanks for confirming :slightly_smiling_face: 

@shaunabanana does this happen every time you launch/start and instance, or only the first time? We did not observe this ourselves. Is there perhaps a setting, do you know? Also, if you could try the same thing after restart, we'd be grateful. We hope the new approach is better, but if it becomes naggy for people, we would like to know. Thank you.

**Comment by ricab** [comment]
Also, credit and thanks to @vsarunas for pointing us toward `arp` :handshake:

**Comment by digows** [comment]
Working like a charm, @ricab 


> 2024-09-20T09:08:09.232] [info] [general] mac_address value is : 52:54:00:25:fa:8a
> [2024-09-20T09:08:09.232] [warning] [general] before arp_process->execute()
> [2024-09-20T09:08:09.239] [debug] [arp] [23565] started: arp -an
> [2024-09-20T09:08:09.248] [info] [general] arp_exit_state.completed_successfully() value is : true
> [2024-09-20T09:08:09.248] [info] [general] arp_ouput_stream.toStdString() value is : ? (192.168.1.1) at 48:a9:8a:c:a0:14 on en0 ifscope [ethernet]
> ? (192.168.1.17) at 2e:c9:c8:a6:e8:1d on en0 ifscope [ethernet]
> ? (192.168.1.40) at bc:d0:74:1b:c7:d on en0 ifscope permanent [ethernet]
> ? (192.168.1.57) at ca:e1:68:e3:4d:e3 on en0 ifscope [ethernet]
> ? (192.168.1.100) at 84:39:be:63:ea:d2 on en0 ifscope [ethernet]
> ? (192.168.1.255) at ff:ff:ff:ff:ff:ff on en0 ifscope [ethernet]
> ? (192.168.64.4) at 52:54:0:25:fa:8a on bridge100 ifscope [bridge]
> ? (224.0.0.251) at 1:0:5e:0:0:fb on en0 ifscope permanent [ethernet]
> 
> [2024-09-20T09:08:09.371] [debug] [via-fronteira] Caching initial SSH session

Thank you so much!

**Comment by vsarunas** [comment]
Thanks @ricab ! Works for me now out of the box now!

`arp -an` triggers [dump(0)](https://github.com/apple-oss-distributions/bootp/blob/rel/bootp-455/bootplib/arp.c#L357-L370) which does some sysctl mib calls to retrieve the arp table; might be more future proof in the unlikely format change in the future.

I'm also waiting for macOS 15 to appear on https://opensource.apple.com/releases/ or the mentioned repo to be able to understand what was done to hide the arp address (changing the address in `/var/db/dhcpd_leases` does cause the instance to actually get that address).

**Comment by shaunabanana** [comment]
> Great, thanks for confirming 🙂
> 
> @shaunabanana does this happen every time you launch/start and instance, or only the first time? We did not observe this ourselves. Is there perhaps a setting, do you know? Also, if you could try the same thing after restart, we'd be grateful. We hope the new approach is better, but if it becomes naggy for people, we would like to know. Thank you.

I tried both, and the prompt for permission did not show up again. Seems to be a one-off thing so it should be fine :) Thank you again for your awesome work!

**Comment by ricab** [comment]
> 

> `arp -an` triggers [dump(0)](https://github.com/apple-oss-distributions/bootp/blob/rel/bootp-455/bootplib/arp.c#L357-L370) which does some sysctl mib calls to retrieve the arp table

Yeah, I noticed that in `dtrace` output :)

> Might be more future proof in the unlikely format change in the future.

My thoughts exactly. I have hopes that this will take care of a fraction of the timeout/unknown people report.

> I'm also waiting for macOS 15 to appear on https://opensource.apple.com/releases/ or the mentioned repo to be able to understand what was done to hide the arp address (changing the address in /var/db/dhcpd_leases does cause the instance to actually get that address).

Yep, it might even be a bug. Before it had the MAC in both the `hw_address` and `identifier` fields, now it has what appears to be an id in both, so maybe they introduced a bug by trying to fix another? We looked a little but didn't find a mapping between those 2 things, not indication that the strange ID could encode the MAC address.

Thanks for sharing that. We'd looked a little into the latest source but found nothing helpful for `dhcpd_leases`. Then you suggested `arp` :)

**Comment by nayneenarh** [comment]
It worked. @ricab 😭 💃🏼💃🏼

**Comment by skywatchinfo** [comment]
same problem on both amd64 and arm64 with  MacOS Sequoia 15.0 (24A335)

**Comment by jsilff** [comment]
The test package worked for me as well, thank you for your speedy work!

**Comment by skywatchinfo** [comment]
test package worked for me . thanks

**Comment by sreekanthjnayar** [comment]
The test package worked for me. I did face a problem while I was in the process of installing the test package where the multipass daemon was still running inspite of uninstalling my older version of the app. I located the daemon plist file under /Library/LaunchDaemons/ using sudo find /Library/LaunchDaemons/ -name "*<app_name>*" - first unloaded it and then deleted it. Post that I was able to get the primary running using the test package.

**Comment by Anubhav-Ray** [comment]
Thank you so much. It is now working.

**Comment by khalid3bdallah** [comment]
The test package works for me
Really appreciate working so quickly to to fix this issue 🙏🏻 🤍

**Comment by HansS1964** [comment]
The test package was a perfect fix. Thanks for the quick effort and resolution. When do you expect this release to be public distributed ?

**Comment by ricab** [comment]
Hi @HansS1964, I am glad it worked for you. If we have no further surprises, we're aiming to have a release candidate (RC) ready in *approximately* a couple of weeks. After that, if all is well with the RC it still takes a few days to sign and finally release.

**Comment by vsarunas** [comment]
Thanks, @ricab!

Given the significant impact and the increasing number of people encountering this issue, would it be worthwhile to consider releasing a 1.14.1 version with just the ARP adjustment to bypass the usual release management processes?

**Comment by ricab** [comment]
Hey @vsarunas, all. I understand that this is a hassle and everyone's desire for a speedy fix. That is why we prioritized this. However, having just this fix would unfortunately not allow us to bypass much. All the other fixes are in and this issue is the only reason why we didn't kick off the patch release earlier.

Beyond that, we prefer not to cut any corners. That would only increase the chances of introducing new problems. All the network issue reports in macOS Sequoia are a good reminder of that to us. It very much looks like a premature release.

The test package is out to unblock anyone affected by this, but do let us know if the link expires. Please bear with us for a little while longer and thank you for your patience!

**Comment by vijaynarayanan-it** [comment]
Thanks for the update @ricab

**Comment by georgeliao** [comment]
> % multipass shell     
> shell failed: Cannot retrieve credentials in unknown state

Sorry that you have encountered this. Can you show the state of the VM first via `multipass list`? Was it running? If it is in the `unknown` state, can you try `multipass stop <vm_name> --force` and multipass start <vm_name>? Can you also launch a new VM?

**Comment by matejk** [comment]
```
% multipass list
Name                    State             IPv4             Image
xts                     Unknown           --               Ubuntu 22.04 LTS
```

`multipass stop xts --force` stopped the VM successfully.

`multipass start xts' shows spinning indicator `Starting xts \`, but does not finish.

Status of the VM in the Multipass GUI is `unknown`.

**Comment by matejk** [comment]
New instance is loaded, starts, I get a prompt to access the network and then I can use it normally.

It seems that the issue is only with the VM that was created before installing the test package.

**Comment by matejk** [comment]
@georgeliao , is is somehow possible to enable network access for the VM that I had on my computer before the upgrade?

**Comment by ricab** [comment]
Hi @matejk, there are multiple possible causes for the symptom you are observing. It is most likely unrelated to this issue and we'd need more info to help investigate. Could you please open a [new issue](https://github.com/canonical/multipass/issues/new/choose) and fill in the form? Thank you.

**Comment by ralgozino** [comment]
Hi!

I upgraded to Sequoia this weekend and also found that multipass is not working anymore. 

I get the same symptoms as the other commenters. VMs starting but not coming ready and changes in the DHCP leases file.

Using the "test package" linked in the comment https://github.com/canonical/multipass/issues/3661#issuecomment-2363403467 above, my primary instance now starts and I can shell into it (with a different IP address than before by the way, same subnet but the host part increased in one), but it is not mounting the Home folder unfortunately.

I tried restarting the host machine but same issue remains with the Home folder not being mounted.

Sorry for commenting in a closed issue, I did not find a better place for feedback on the test package.

My system is an M1 macbook running macos Sequoia 15.0 (24A335), let me know if I can help with any more info or test

**Comment by ricab** [comment]
Hi @ralgozino, I don't think this is related to new changes. Would you mind opening a new issue and filling in the form with your info? Thanks.

**Comment by ralgozino** [comment]
FWIW, after deleting with `--purge` the primary instance and recreating it, the home folder got mounted again.

**Comment by rachadaccoumeh** [comment]
Hi, I have the same problem here the new test package worked for me after deleting the whole multipass with all previous instances but each instance only works normally for me the first time after creating the instance otherwise I'm not able to connect to it from terminal or GUI some time I'm getting started status but unable to connect to it some time I got unknown status ( i noticed that I'm able to connect to it via ssh) but I'm unable to access the web server via same IP used for ssh.

**Comment by daryush** [comment]
I tried to use test package mentioned above and unfortunately it didn't worked for me. I'm not able to put newly created instance to running state, it tries to start and land in unknown state. It didn't asked me for permission to access the network neither. Below are my logs from my tries

Starting new instance
```
% multipass launch jammy --name bff7 --memory 20G --disk 50G --cpus 10 -vvvv
[2024-09-30T11:42:49.174] [debug] [qemu-system-aarch64] [3606] started: qemu-system-aarch64 --version
[2024-09-30T11:42:49.464] [debug] [qemu-img] [3608] started: qemu-img info /var/root/Library/Caches/multipassd/qemu/vault/images/jammy-20240912/ubuntu-22.04-server-cloudimg-arm64.img
[2024-09-30T11:42:49.487] [trace] [client cert store] Verifying cert:
-----BEGIN CERTIFICATE-----
MIIBoDCCAUegAwIBAgIEWvI8MzAKBggqhkjOPQQDAjBQMQswCQYDVQQGEwJVUzES
MBAGA1UECgwJQ2Fub25pY2FsMS0wKwYDVQQDDCQxNzBiMzE4Ni04NTA2LTQ1YTAt
OWFkYS0wODBjYjg4Mjk4MWQwHhcNMjQwNzA0MTIyMTUxWhcNMjUwNzA0MTIyMTUx
WjBQMQswCQYDVQQGEwJVUzESMBAGA1UECgwJQ2Fub25pY2FsMS0wKwYDVQQDDCQx
NzBiMzE4Ni04NTA2LTQ1YTAtOWFkYS0wODBjYjg4Mjk4MWQwWTATBgcqhkjOPQIB
BggqhkjOPQMBBwNCAASjxdhdpj1sXGLnmMTp7Dq3s/OklyOzNWOkCFY6ZiMvIEvo
oeNTR+/yDAhavjwZlTimiz8mKJx/DeWJd/c1krXgow8wDTALBgNVHREEBDACggAw
CgYIKoZIzj0EAwIDRwAwRAIgPSHQMG+d2o5ABLNACDMvQrqP6Oex9EHpqGSjeLmd
CKcCIBv+kE7LqEdVSIP5pVghIHfQmXvr6bwHPIFmhVf9FH0Q
-----END CERTIFICATE-----

[2024-09-30T11:42:49.739] [debug] [qemu-img] [3609] started: qemu-img resize /var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/ubuntu-22.04-server-cloudimg-arm64.img 53687091200
[2024-09-30T11:42:49.757] [debug] [qemu-img] [3610] started: qemu-img snapshot -l /var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/ubuntu-22.04-server-cloudimg-arm64.img
[2024-09-30T11:42:49.768] [debug] [qemu-img] [3611] started: qemu-img amend -o compat=1.1 /var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/ubuntu-22.04-server-cloudimg-arm64.img
[2024-09-30T11:42:49.779] [debug] [bff7] process working dir ''
[2024-09-30T11:42:49.779] [info] [bff7] process program 'qemu-system-aarch64'
[2024-09-30T11:42:49.779] [info] [bff7] process arguments '-machine, virt,gic-version=3, -accel, hvf, -drive, file=/Library/Application Support/com.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on, -cpu, host, -nic, vmnet-shared,model=virtio-net-pci,mac=52:54:00:f9:81:d0, -device, virtio-scsi-pci,id=scsi0, -drive, file=/var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/ubuntu-22.04-server-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda, -device, scsi-hd,drive=hda,bus=scsi0.0, -smp, 10, -m, 20480M, -qmp, stdio, -chardev, null,id=char0, -serial, chardev:char0, -nographic, -cdrom, /var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/cloud-init-config.iso'
[2024-09-30T11:42:49.784] [debug] [qemu-system-aarch64] [3612] started: qemu-system-aarch64 -machine virt,gic-version=3 -nographic -dump-vmstate /private/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/multipassd.zfGOkg
[2024-09-30T11:42:50.101] [info] [bff7] process state changed to Starting
[2024-09-30T11:42:50.106] [info] [bff7] process state changed to Running
[2024-09-30T11:42:50.106] [debug] [qemu-system-aarch64] [3613] started: qemu-system-aarch64 -machine virt,gic-version=3 -accel hvf -drive file=/Library/Application Support/com.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on -cpu host -nic vmnet-shared,model=virtio-net-pci,mac=52:54:00:f9:81:d0 -device virtio-scsi-pci,id=scsi0 -drive file=/var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/ubuntu-22.04-server-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda -device scsi-hd,drive=hda,bus=scsi0.0 -smp 10 -m 20480M -qmp stdio -chardev null,id=char0 -serial chardev:char0 -nographic -cdrom /var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/cloud-init-config.iso
[2024-09-30T11:42:50.106] [info] [bff7] process started
launch failed: The following errors occurred:
bff7: timed out waiting for response
```

I've tried to stop it with force and start once again and also it didn't worked
```
% multipass stop bff7 --force


% multipass start bff7 -vvvv 
[2024-10-02T11:05:12.766] [debug] [bff7] process working dir ''
[2024-10-02T11:05:12.767] [info] [bff7] process program 'qemu-system-aarch64'
[2024-10-02T11:05:12.769] [info] [bff7] process arguments '-machine, virt,gic-version=3, -accel, hvf, -drive, file=/Library/Application Support/com.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on, -cpu, host, -nic, vmnet-shared,model=virtio-net-pci,mac=52:54:00:f9:81:d0, -device, virtio-scsi-pci,id=scsi0, -drive, file=/var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/ubuntu-22.04-server-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda, -device, scsi-hd,drive=hda,bus=scsi0.0, -smp, 10, -m, 20480M, -qmp, stdio, -chardev, null,id=char0, -serial, chardev:char0, -nographic, -cdrom, /var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/cloud-init-config.iso'
[2024-10-02T11:05:12.901] [debug] [qemu-system-aarch64] [43846] started: qemu-system-aarch64 -machine virt,gic-version=3 -nographic -dump-vmstate /private/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/multipassd.PtMYuq
[2024-10-02T11:05:12.970] [info] [bff7] process state changed to Starting
[2024-10-02T11:05:13.022] [info] [bff7] process state changed to Running
[2024-10-02T11:05:13.022] [debug] [qemu-system-aarch64] [43847] started: qemu-system-aarch64 -machine virt,gic-version=3 -accel hvf -drive file=/Library/Application Support/com.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on -cpu host -nic vmnet-shared,model=virtio-net-pci,mac=52:54:00:f9:81:d0 -device virtio-scsi-pci,id=scsi0 -drive file=/var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/ubuntu-22.04-server-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda -device scsi-hd,drive=hda,bus=scsi0.0 -smp 10 -m 20480M -qmp stdio -chardev null,id=char0 -serial chardev:char0 -nographic -cdrom /var/root/Library/Application Support/multipassd/qemu/vault/instances/bff7/cloud-init-config.iso
[2024-10-02T11:05:13.022] [info] [bff7] process started
start failed: The following errors occurred:
bff7: timed out waiting for response
```

My version is installed from the comment above
```
% multipass version
multipass   1.15.0-dev.2929.pr661+gc67ef6641.mac
multipassd  1.15.0-dev.2929.pr661+gc67ef6641.mac
```

I work on MacBook Pro with M1 and Sonoma 14.5

**Comment by levkropp** [comment]
Hi @rachadaccoumeh ,

Since this issue is closed, could you please [open a new issue](https://github.com/canonical/multipass/issues/new/choose) and fill out the issue template? This will help us get all the necessary information to troubleshoot effectively. Thank you!

**Comment by ricab** [comment]
Interesting @vsarunas, thanks for the news! We've had one report last week of someone who was on 15 but did not see the change in `dhcpd_leases`, so I wonder if there is some other condition necessary to trigger the change/bug.

In any case, reading from that file was sometimes unreliable (old IPs in the mix and such). So we'll try the ARP approach now and see how it fares for people.

As an update: the release packages have been *mostly* ready for a few days now, but we are waiting for them to get signed internally. Unfortunately, there are ongoing issues with the hardware, so it is taking longer than usual. You can find (unsigned) RCs with a number of other bug fixes [here](https://github.com/canonical/multipass/releases/tag/v1.14.1-rc1).

**Comment by vsarunas** [comment]
In case there are issues to install the RC1 and verification warnings:
```
Apple could not verify “multipass-1.14.1-rc1+mac.14+gf2381bfe9.mac-Darwin.pkg” is free of malware that may harm your Mac or compromise your privacy.
```

Need to do:
```
xattr -d com.apple.quarantine  ~/Downloads/multipass-1.14.1-rc1+mac.14+gf2381bfe9.mac-Darwin.pkg
```

**Comment by niluan-zwl** [comment]
oday after upgrading to macOS Sequoia 15.01  all instances in multipass cannot be started and cannot obtain an IP address.

**Comment by jakub-zawislak** [comment]
I have two instances in Multipass. I installed the test package of Multipass a few days ago. After the installation, only one instance was working. Today, the previously working instance instance also crashed. 

In both instances, I have logs like these:

```
[debug] [primary] process working dir ''
[info] [primary] process program 'qemu-system-aarch64'
[info] [primary] process arguments '-machine, virt,gic-version=3, -accel, hvf, -drive, file=/Library/Application Support/com.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on, -cpu, host, -nic, vmnet-shared,model=virtio-net-pci,mac=52:54:00:6b:21:83, -device, virtio-scsi-pci,id=scsi0, -drive, file=/var/root/Library/Application Support/multipassd/qemu/vault/instances/primary/ubuntu-22.04-server-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda, -device, scsi-hd,drive=hda,bus=scsi0.0, -smp, 4, -m, 7168M, -qmp, stdio, -chardev, null,id=char0, -serial, chardev:char0, -nographic, -cdrom, /var/root/Library/Application Support/multipassd/qemu/vault/instances/primary/cloud-init-config.iso'
[debug] [qemu-system-aarch64] [647] started: qemu-system-aarch64 -machine virt,gic-version=3 -nographic -dump-vmstate /private/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/multipassd.OfALEJ
[info] [primary] process state changed to Starting
[info] [primary] process state changed to Running
[debug] [qemu-system-aarch64] [649] started: qemu-system-aarch64 -machine virt,gic-version=3 -accel hvf -drive file=/Library/Application Support/com.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,format=raw,readonly=on -cpu host -nic vmnet-shared,model=virtio-net-pci,mac=52:54:00:6b:21:83 -device virtio-scsi-pci,id=scsi0 -drive file=/var/root/Library/Application Support/multipassd/qemu/vault/instances/primary/ubuntu-22.04-server-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda -device scsi-hd,drive=hda,bus=scsi0.0 -smp 4 -m 7168M -qmp stdio -chardev null,id=char0 -serial chardev:char0 -nographic -cdrom /var/root/Library/Application Support/multipassd/qemu/vault/instances/primary/cloud-init-config.iso
[info] [primary] process started
[debug] [primary] Waiting for SSH to be up
[debug] [arp] [650] started: arp -an
[debug] [primary] QMP: {"QMP": {"version": {"qemu": {"micro": 1, "minor": 2, "major": 8}, "package": ""}, "capabilities": ["oob"]}}

[debug] [primary] QMP: {"return": {}}

[debug] [arp] [872] started: arp -an
[debug] [arp] [893] started: arp -an
[debug] [arp] [1035] started: arp -an
[debug] [arp] [1045] started: arp -an
```

I already rebooted macOS a few times, I was also stopping instances with `--force` or killing the `qemu` process.
I have no issues with creating new instances.

**Comment by stuartskelton** [comment]
Hi

I just wanted to add the  test package  worked on my newly updated and fresh 15.0.

Regards

Stuart

**Comment by michaeldam** [comment]
1.14.1 RC1 works on M1 running on fresh MacOS 15.0.1

Obviously needed the xattr to be able to install: https://github.com/canonical/multipass/issues/3661#issuecomment-2396933827

**Comment by ricab** [comment]
Hi @sekiskylink, that indicates you have some other problem with your instance. In order for us to help you, we'd need a [new issue](https://github.com/canonical/multipass/issues/new/choose) with the requested info, please.

**Comment by ricab** [comment]
Hi all, just wanted to let everyone involved here know that we finally got the signed packages and released [1.14.1](https://github.com/canonical/multipass/releases/tag/v1.14.1) officially. Apologies for the delay.

**Comment by khalid3bdallah** [comment]
@ricab 
My current version:
```
multipass   1.15.0-dev.2929.pr661+gc67ef6641.mac
multipassd  1.15.0-dev.2929.pr661+gc67ef6641.mac
```

Are there specific steps I need to follow to install version 1.14.1?

**Comment by ricab** [comment]
Hi @khalid3bdallah, not really, you should just be able to download the package from the link above and double click it. The version looks lower, but it is not, because yours is just a dev package (toward 1.15).

The change to multipass.run is in progress too, so people will be able to get it from there too in the next few days.

**Comment by SOFTWORK** [comment]
Has anyone the Test Package File multipass-1.15.0-dev.2929.pr661+gc67ef6641.mac-Darwin.pkg - the link provided here is dysfunctional - many thanks.

**Comment by puneetdawer** [comment]
I am not able to access test package and it gives access denied error.

**Comment by tomsonpl** [comment]
Unfortunately I am still seeing the issues even after upgrading to `1.14.1`.

```
$ multipass shell test
shell failed: ssh connection failed: 'Failed to connect: No route to host'
```

It worked only after using these:
```
multipass set local.passphrase
sudo multipass authenticate
sudo multipass shell test
```

**Comment by ricab** [comment]
That's reported in #3766 @tomsonpl .

**Comment by alexcardo** [comment]
So, today is 5th december. I've spent the entire day in an attempt to launch Multipass. 

<img width="1396" alt="Снимок экрана 2024-12-05 в 17 46 22" src="https://github.com/user-attachments/assets/cf1126b6-1cf2-485b-a9a7-b70aae3f9339">
This lasts forever ... and no solution...

**Comment by ricab** [comment]
That is unrelated to this issue @alexcardo. Would you mind opening a new one, filling in the information in the form? Thank you.
